integration_tests/src/main/python/spark_session.py

```diff
@@ -83,7 +83,12 @@ def reset_spark_session_conf():
 def _check_for_proper_return_values(something):
     """We don't want to return an DataFrame or Dataset from a with_spark_session. You will not get what you expect"""
     if (isinstance(something, DataFrame)):
-        raise RuntimeError("You should never return a DataFrame from a with_*_session, you will not get the results that you expect")
+        error_msg = """
+        You are trying to return a DataFrame from a with_*_session.
+        This is not allowed because the dataframe is processed lazily,
+        leading to discrepancies between the actual behavior and what you might expect.
```

Comment:
for the source code it looks better but we now end up with the log that is more difficult to parse because of the newlines. Since in burndown, can you undo unrelated changes?

Reply:
I reverted this change as I understand this is out of scope. My comment was that I disagree with the new error message is harder to parse.